tests/Node.py

```diff
@@ -1424,16 +1424,50 @@ def isLibAdvancing():
             return self.getIrreversibleBlockNum() > currentLib
         return Utils.waitForTruth(isLibAdvancing, timeout)
 
+    def waitUntilBeginningOfProdTurn(self, producerName, timeout=30, sleepTime=0.4):
+        beginningOfProdTurnHead = 0
+        def isDesiredProdTurn():
+            beginningOfProdTurnHead = self.getHeadBlockNum()
```

Comment:
This updates a local copy of `beginningOfProdTurnHead`, leaving the variable of the same name in the parent scope unchanged.

Reply:
I changed it to return the value, and only set the variable where it was being used, it was never used after that, so I didn't reset the variable on subsequent calls

Reply:
I missed what you were meaning here, rather than in the original place it was being used.  Fixed it now.